apps/prairielearn/src/ee/auth/lti13/lti13Auth.html.ts

```diff
@@ -47,6 +46,43 @@ export function Lti13Test({
   `.toString();
 }
 
+export function Lti13AuthIframe({ parameters }: { parameters: Record<string, any> }) {
+  return html`
+    <!doctype html>
+    <html lang="en">
+      <head>
+        ${HeadContents({ resLocals: {}, pageTitle: 'LTI 1.3 redirect' })}
+      </head>
+      <body>
+        <main id="content" class="m-3">
+          <form id="interceptForm" method="POST" action="" target="_blank">
+            ${Object.entries(parameters).map(
+              ([key, value]) =>
+                html` <input type="hidden" name="${key}" value="${String(value)}" />`,
+            )}
+            <button id="submitButton" class="btn btn-primary btn-lg">
+              Open PrairieLearn in a new window
+            </button>
+          </form>
+          <div id="message">If there are login errors, reload this page to try again.</div>
```

Comment:
What problem does this aim to solve? It's not clear why reloading this page would do anything different.

Reply:
One error case I can think of would be loading the page (button in the iframe) and not pressing it, waiting until some  timeframe expires (which isn't visible to the user), and then pressing the button could produce an error when either PL or the LMS processes the LTI launch. In a normal workflow, we're expecting people click through quickly and this doesn't happen but if it does, we don't have a way to soften the error so I was pre-warning.

The other case I found when developing was reusing the button was definitely going to cause an error, which usually were ugly and not well described. Using JS to disable the button, like Canvas does with its "launch in a new window" functionality, protects against this.

Reply:
Ahhh, right - "this page" in this context is going to be the LMS, right? I think that was the piece that didn't click for me til now. It might be worth leaving a small comment to emphasize that fact?

100% agree with disabling the button after it's clicked.

It feels a little weird to me that "Reload the page to try again." shows up after the button is clicked. That seems like the kind of thing that would be shown if something _did_ go wrong. I think maybe leaving "If there are login errors..." as the text might be best?

Reply:
For Canvas's version of this functionality, they make the button disappear after it's pressed and use this text: "This tool was successfully loaded in a new browser window. Reload the page to access the tool again."

Reply:
The wording "try again" is specifically what feels wrong to me. I would not expect to see "try again" if something succeeded. I always see that the context of something going wrong, like "Your payment method was not processed, please try again".

Reply:
We can avoid "try again". My current draft has these lines (the after line crafted from the Canvas message):

Before: If there are login errors, reload this page to start again.
After: Reload this page to access PrairieLearn again.